packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -455,6 +455,13 @@ class IrisGridTableModelTemplate<
     return !this.isSaveInProgress && this.inputTable != null;
   }
 
+  get isDeletable(): boolean {
+    if (this.inputTable !== null) {
+      return this.inputTable?.keyColumns.length > 0;
+    }
+    return false;
+  }
+
```

Comment:
Ah I see what's missing. At the top, we have `IrisGridTableModelTemplate` which `extends IrisGridModel`, but we should also specify that it `implements DeletableGridModel, EditableGridModel`. That way we can be sure the types of each method match correctly, and we're not missing any methods.
Add that in, you may need to adjust some of the types (mostly just adding a `readonly` in front of arrays).